pkg/apis/messaging/v1beta1/kafka_channel_lifecycle.go

```diff
@@ -99,8 +99,18 @@ func (kcs *KafkaChannelStatus) GetCondition(t apis.ConditionType) *apis.Conditio
 }
 
 // IsReady returns true if the resource is ready overall.
-func (kcs *KafkaChannelStatus) IsReady() bool {
-	return kcs.GetConditionSet().Manage(kcs).IsHappy()
+func (kc *KafkaChannel) IsReady() bool {
+	kcs := kc.Status
+	return kcs.ObservedGeneration == kc.Generation &&
+		kcs.GetCondition(KafkaChannelConditionReady).IsTrue()
```

Comment:
```suggestion
     kcs.GetConditionSet().Manage(kcs).IsHappy()
```
I know Serving isn't using this, but IMHO they should. There was a debate about it where folks didn't wanna stop the PR for, but in our case unless someone has a strong opinion why we shouldn't use `.IsHappy()` then I think we should use the ConditionsManager as it's intended to be used.

Reply:
Thanks for the heads up. I will go w/ `Happy` than. Did not know the context .

Reply:
@devguyio 

```
cannot use kcs (variable of type KafkaChannelStatus) as apis.ConditionsAccessor value in argument to kcs.GetConditionSet().Manage: missing method GetConditions (GetConditions has pointer receiver)
```

I think that's a reason, at least for me.

Reply:
@devguyio  OK.   one more commit to this PR:


https://github.com/knative-sandbox/eventing-kafka/pull/825/commits/fa2ebcbfc55c8fd58aa6160b41762383ea93bd54